test/components/translation_nav_test.rb

```diff
@@ -0,0 +1,82 @@
+require 'component_test_helper'
+
+class TranslationNavTest < ComponentTestCase
+  def component_name
+    "translation-nav"
+  end
+
+  test "fails to render a translation nav when no translations are given" do
+    assert_raise do
+      render_component({})
+    end
+  end
+
+  test "renders an active translation nav item with a text description" do
+    render_component(
+      translations: [
+        {
+          locale: 'en',
+          base_path: '/en',
+          text: 'English',
+          active: true
+        },
+        {
```

Comment:
We need to have assertion for all the translations, otherwise we could break this and the tests would still pass.